src/js/app.js

```diff
@@ -0,0 +1,65 @@
+$('document').ready(function() {
+
+  //findPics wittles down an object to an array of desired data
+  //by using .filter() to filter out objects that don't meet a certain requirement
+  //and by using .map()  to transform the filtered object in to an array of desired data
+  var findPics = function(dataObj) {
+    var filterResult = dataObj.data.children.filter(function(element) {
+      if (element.data.hasOwnProperty('preview')) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+
+    var urlArray = filterResult.map(function(element) {
+      return element.data.preview.images[0].source.url;
+    });
+
+    var index = 0;
+    intervalID = window.setInterval(function() {
+      index++;
+      if (index >= urlArray.length) {
+        index = 0;
+      }
+
+      $('#results').html('<img src="' + urlArray[index] + '">');
+    }, 1000);
+  };
+
+  //text inpout event listener that detects the enter key and
+  // registers a click event on the button
+
+  $('#textForSearch').keyup(function(event) {
+    if (event.keyCode === 13) {
+      $('#buttonForSearch').click();
+    }
+  });
```

Comment:
Nice job using the keycodes. It's really handy for games too. Another alternative would be to add a `submit` handler on the form, so you would only need one event handler.